packages/phone-number-privacy/combiner/test/integration/pnp.test.ts

```diff
@@ -0,0 +1,736 @@
+      let req: SignMessageRequest
+
+      beforeEach(() => {
+        mockOdisPaymentsTotalPaidCUSD.mockReturnValue(onChainPaymentsDefault)
+        req = getSignRequest(blindedMsgResult)
+      })
+
+      it('Should respond with 500 even if request is valid', async () => {
+        const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+        const res = await sendPnpSignRequest(req, authorization, app)
+
+        expect(res.status).toBe(500)
+        expect(res.body).toMatchObject<SignMessageResponseFailure>({
+          success: false,
+          version: expectedVersion,
+          error: ErrorMessage.NOT_ENOUGH_PARTIAL_SIGNATURES,
+        })
+      })
+    })
+  })
+
+  describe(`${CombinerEndpoint.PNP_QUOTA}`, () => {
+    const useQuery = async (queryCount: number, signer: Server | HttpsServer) => {
+      for (let i = 0; i < queryCount; i++) {
+        const phoneNumber = '+1' + Math.floor(Math.random() * 10 ** 10)
+        const blindedNumber = getBlindedPhoneNumber(phoneNumber, BLINDING_FACTOR)
+        const req = {
+          account: ACCOUNT_ADDRESS1,
+          blindedQueryPhoneNumber: blindedNumber,
+        }
+        const authorization = getPnpRequestAuthorization(req, PRIVATE_KEY1)
+        await request(signer)
+          .post(SignerEndpoint.PNP_SIGN)
+          .set('Authorization', authorization)
+          .send(req)
+      }
+    }
+
+    const getCombinerQuotaResponse = async (req: PnpQuotaRequest, authorization: string) => {
+      const res = await request(app)
+        .get(CombinerEndpoint.PNP_QUOTA)
+        .set('Authorization', authorization)
+        .send(req)
+      return res
+    }
+
+    const totalQuota = 10
+    beforeAll(async () => {
+      let weiTocusd = new BigNumber(1e17)
+      mockOdisPaymentsTotalPaidCUSD.mockReturnValue(weiTocusd.multipliedBy(totalQuota))
+    })
+
+    beforeEach(async () => {
+      signer1 = startSigner(signerConfig, signerDB1, keyProvider1, mockKit).listen(3001)
+      signer2 = startSigner(signerConfig, signerDB2, keyProvider2, mockKit).listen(3002)
+      signer3 = startSigner(signerConfig, signerDB3, keyProvider3, mockKit).listen(3003)
+    })
+
+    const queryCountParams = [
+      { signerQueries: [0, 0, 0], expectedQueryCount: 0 },
+      { signerQueries: [1, 0, 0], expectedQueryCount: 0 }, // does not reach threshold
+      { signerQueries: [1, 1, 0], expectedQueryCount: 1 }, // threshold reached
+      { signerQueries: [0, 1, 1], expectedQueryCount: 1 }, // order of signers shouldn't matter
+      { signerQueries: [1, 4, 9], expectedQueryCount: 4 },
+    ]
+    queryCountParams.forEach(({ signerQueries, expectedQueryCount }) => {
+      it(`should get ${expectedQueryCount} performedQueryCount given signer responses of ${signerQueries}`, async () => {
+        await useQuery(signerQueries[0], signer1)
+        await useQuery(signerQueries[1], signer2)
+        await useQuery(signerQueries[2], signer3)
+
+        const req = {
```

Comment:
(optional nit): consider using the `getPnpQuotaRequest` method (or update as needed) from `TestUtils`